example-todomvc/src/main/scala/mhtml/todo/MhtmlTodo.scala

```diff
@@ -0,0 +1,342 @@
+import scala.collection.breakOut
+import cats._
+import cats.data._
+import cats.implicits._
+import mhtml._
+import mhtml.implicits.cats._
+import org.scalajs.dom
+import org.scalajs.dom.{Event, HashChangeEvent, KeyboardEvent}
+import org.scalajs.dom.ext.KeyCode
+import org.scalajs.dom.ext.LocalStorage
+import org.scalajs.dom.raw.HTMLInputElement
+import upickle.default.read
+import upickle.default.write
+
+import scala.scalajs.runtime.UndefinedBehaviorError
+
+object MhtmlTodo extends JSApp {
+
+  case class Component[D](view: Node, model: Rx[D])
+
+  class Todo(val title: String, val completed: Boolean)
+  object Todo {
+    def apply(title: String, completed: Boolean) = new Todo(title, completed)
+    def unapply(todo: Todo) = Option((todo.title, todo.completed))
+  }
+
+  case class TodoList(text: String, hash: String, items: Rx[Seq[Todo]])
+
+  //
+  // Add components for the app
+  //
+
+  lazy val header: Component[Option[Todo]] = {
+    val newTodo = Var[Option[Todo]](None)
+    val onInputKeydown: (KeyboardEvent) => Unit = { event: KeyboardEvent =>
+      (event.currentTarget, event.keyCode) match {
+        case (input: HTMLInputElement, KeyCode.Enter) =>
+          input.value.trim match {
+            case "" =>
+            case title =>
+              newTodo := Some(Todo(title, completed = false))
+              input.value = ""
+          }
+        case _ =>
+      }
+    }
+    val headerNode =
+      <header class="header">
+        <h1>todos</h1>
+        <input class="new-todo"
+               autofocus="true"
+               placeholder="What needs to be done?"
+               onkeydown={onInputKeydown}/>
+      </header>
+    Component(headerNode, newTodo)
+  }
+
+  def footer: Component[List[Todo]] = {
+    val removeTodos = Var[List[Todo]](Nil)
+    val display = allTodos.map(x => if (x.isEmpty) "none" else "")
+    val visibility =
+      completed.items.map(x => if (x.isEmpty) "hidden" else "visible")
+    val footerNode =
+      <footer class="footer" style:display={display}>
+        <ul class="filters">{todoLists.map(todoListsFooter)}</ul>
+        <button onclick={() =>
+            allTodos.map(_.filter(_.completed).foreach(todo =>
+              removeTodos.update(rtList => todo :: rtList)
+            ))
+            ()
+          }
+          class="clear-completed"
+          style:visibility={visibility}>
+          Clear completed
+        </button>
+      </footer>
+    Component(footerNode, removeTodos)
+  }
+
+  case class TodoUpdate(oldTodo: Todo, newTodo: Todo)
+  case class TodoListItemData(removal: Option[Todo], update: Option[TodoUpdate])
+  //
+  def todoListItem(todo: Todo): Component[TodoListItemData] = {
+    val removeTodo = Var[Option[Todo]](None)
+    val updateTodo = Var[Option[TodoUpdate]](None)
+    val suppressOnBlur = Var(false)
+    def submit: Event => Unit = { event: Event =>
+      suppressOnBlur := true
+      editingTodo := None
```

Comment:
Global `Var` :confused: Is there a way around that without introducing a `Component2`?